shm/player.cpp

```diff
@@ -0,0 +1,33 @@
+#include "player.hpp"
+
+#include <numeric>
+
+Player::Player(std::unique_ptr<Ship> ship, size_t money, size_t availableSpace)
+    : ship_(std::move(ship)), money_(money), availableSpace_(availableSpace) {}
+
+std::unique_ptr<Ship> Player::getShip() {
+    return std::move(ship_);
```

Comment:
Are you absolutely sure about that operation? We are not at this level in this course :)